Clean up non-C++14 code

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28443 Clean up non-C++14 code**

We're now on C++14, so we don't need the else branch of these ifdef's anymore

Differential Revision: [D18069136](https://our.internmc.facebook.com/intern/diff/D18069136/)

smessmer committed:
Clean up non-C++14 code

We're now on C++14, so we don't need the else branch of these ifdef's anymore

Differential Revision: [D18069136](https://our.internmc.facebook.com/intern/diff/D18069136/)

[ghstack-poisoned]

VitalyFedyunin commented on aten/src/ATen/cpu/vec256/vec256.h:
```diff
@@ -145,27 +145,14 @@ inline interleave2<double>(const Vec256<double>& a, const Vec256<double>& b) {
   //   a_swapped = {a0, a1, b0, b1}
   //   b_swapped = {a2, a3, b2, b3}
-#if __cpp_binary_literals >= 201304L
   auto a_swapped = _mm256_permute2f128_pd(a, b, 0b0100000);
```
Please put computation formula as comment.

smessmer committed:
Update on "Clean up non-C++14 code"


We're now on C++14, so we don't need the else branch of these ifdef's anymore

Differential Revision: [D18069136](https://our.internmc.facebook.com/intern/diff/D18069136/)

[ghstack-poisoned]

smessmer committed:
Update on "Clean up non-C++14 code"


We're now on C++14, so we don't need the else branch of these ifdef's anymore

Differential Revision: [D18069136](https://our.internmc.facebook.com/intern/diff/D18069136/)

[ghstack-poisoned]

smessmer committed:
Update on "Clean up non-C++14 code"


We're now on C++14, so we don't need the else branch of these ifdef's anymore

Differential Revision: [D18069136](https://our.internmc.facebook.com/intern/diff/D18069136/)

[ghstack-poisoned]

smessmer committed:
Update on "Clean up non-C++14 code"


We're now on C++14, so we don't need the else branch of these ifdef's anymore

Differential Revision: [D18069136](https://our.internmc.facebook.com/intern/diff/D18069136/)

[ghstack-poisoned]

ssnl reviewed (approved):
lgtm.
